### ISSUE TITLE
Error in greenplum segment logs

### ISSUE THREAD

**Issue by mialinx** [wal-g/wal-g#1465]
### Greenplum

# Issue description
wal-g-gp tries to connect to the segment not in utility mode.

### Describe your problem
Running wal-g seg-backup-push
```
INFO: 2023/04/18 02:51:21.648841 starting the command: /usr/bin/wal-g seg-backup-push --content-id=0 /var/lib/greenplum/data1/primary/gpseg0 --add-user-data={"id":"c0a6182d-dc99-4cac-85a0-c5f8f678da50"} --pgport=6000 --permanent=false --config /etc/wal-g/wal-g.yaml
```
causes messages in postgress  log  like that
```
2023-04-18 02:51:21.691735 MSK,"gpadmin","postgres",p3273070,th1770055552,"::1","51942",2023-04-18 02:51:21 MSK,0,,,seg0,,,,sx1,"FATAL","57P03","System was started in master-only utility mode - only utility mode connections are allowed",,,,,,,0,,"postinit.c",1203,
```
(But backup works fine)

### Please provide steps to reproduce
just make a backup and see logs

**Comment by usernamedt** [comment]
Hi! Thanks for your report. The root cause is located [somewhere here](https://github.com/wal-g/wal-g/blob/8e3c05e133e7305d6ec5481870f90bef9a002297/internal/databases/postgres/connect.go#L29). Connection estabilishment code is common for PostgreSQL and GreenplumDB and definetly 
deserves some refactoring. Any PRs addressing this problem would be welcome :)

**Comment by anikin-aa** [comment]
Folks, hi

Actually, for the current codebase, this one does not seem like a bug, because the current logic tries to connect w/o gp specific runtime parameters and, if fails tries to add them. 

If nobody is working on this right now, I can try to improve the connection establishment logic.

**Comment by usernamedt** [comment]
> Folks, hi
> 
> Actually, for the current codebase, this one does not seem like a bug, because the current logic tries to connect w/o gp specific runtime parameters and, if fails tries to add them.
> 
> If nobody is working on this right now, I can try to improve the connection establishment logic.

Yes, your understanding is absolutely correct. AFAIK, no one is currently working on this one. Your contribution would be very welcome!
